### PR TITLE
Remove `$FAIL` function

### PR DESCRIPTION
**Pull request by jugglinmike**
This function is equivalent to `$ERROR` (which is automatically included
in test environments). Remove the harness file that defines the
function, remove references to the file from test `includes` lists, and
update scripts to instead invoke the `$ERROR` function.

Resolves gh-126

### CODE REVIEW

**Comment by anba** [comment]
The three new files in test/built-ins/TypedArray/from are most likely part of a different PR, right?

And `$PRINT()` should probably be removed, too. As a follow-up of course. See:
- Object/prototype/S15.2.4_A3.js
- Object/prototype/S15.2.4_A4.js
- Object/prototype/hasOwnProperty/S15.2.4.5_A7.js
- ... more

File: test/built-ins/Number/prototype/S15.7.3.1_A3.js [hunk 1]
```diff
@@ -5,13 +5,12 @@
 info: Number.prototype value is not +0
 es5id: 15.7.3.1_A3
 description: Checking value of Number.prototype property
-includes: [$FAIL.js]
 ---*/
 
 //CHECK#1
 try {
   (Number.prototype != 0);
-  $FAIL('#1: "(Number.prototype != 0);" lead to throwing exception. Actual: '+(Number.prototype != 0));
```
Comment (bterlson) [inline test/built-ins/Number/prototype/S15.7.3.1_A3.js]: I imagine this is not going to be too friendly of a reported error as $ERROR will throw, causing the check in the catch block to fail as well. But, seems ok for now, it's not like $FAIL was helping anyway.

**Comment by bterlson** [comment]
I guess remove the excess tests and this is g2g.

**Comment by jugglinmike** [comment]
You are correct, @anba. I guess I'm trying to keep too many things in the air at once.

@bterlson I've removed those three tests and force-pushed a new commit.

**Pull request merged**